docs/proposals/docker-registry-2-4.md

```diff
@@ -0,0 +1,57 @@
+# Support for Docker Registry 2.4
+## Abstract
+
+A proposal to update the docker/distribution to v2.4.0.
+
+## Motivation
+
+Primary goal is to migrate to Schema 2 [1].
```

Comment:
I'd say the primary goal is to add support for [Cross-repository push](https://github.com/docker/distribution/issues/634) which allows to complete layer federation feature and will make the pushes a lot faster.

And we also want [Garbage collection](https://github.com/docker/distribution/pull/1386) for the pruning.